src/server/scripts/Northrend/Naxxramas/boss_maexxna.cpp

```diff
@@ -165,23 +193,49 @@ class npc_webwrap : public CreatureScript
 
     struct npc_webwrapAI : public NullCreatureAI
     {
-        npc_webwrapAI(Creature* creature) : NullCreatureAI(creature) { }
+        npc_webwrapAI(Creature* creature) : NullCreatureAI(creature), victimGUID(ObjectGuid::Empty), visibleTimer(0) { }
 
         ObjectGuid victimGUID;
+        uint32 visibleTimer;
+
+        void InitializeAI() override
+        {
+            me->SetVisible(false);
+        }
 
         void SetGUID(ObjectGuid guid, int32 /*param*/) override
         {
+            if (!guid)
```

Comment:
Yes, cocoon NPC can't be re-assigned a new target after its initial target. One cocoon is spawned per wrap.